includes/service-providers/active_campaign/class-newspack-newsletters-active-campaign.php

```diff
@@ -622,30 +622,42 @@ public function trash( $post_id ) {
 	}
 
 	/**
-	 * Add contact to a list.
+	 * Add contact to a list or update an existing contact.
 	 *
-	 * @param array  $contact      {
-	 *    Contact data.
+	 * @param array        $contact      {
+	 *          Contact data.
 	 *
 	 *    @type string   $email    Contact email address.
 	 *    @type string   $name     Contact name. Optional.
 	 *    @type string[] $metadata Contact additional metadata. Optional.
 	 * }
-	 * @param string $list_id      List to add the contact to.
+	 * @param string|false $list_id      List to add the contact to.
 	 *
-	 * @return bool|WP_Error True if the contact was added or error if failed.
+	 * @return int|WP_Error Contact ID if the contact was added or error if failed.
```

Comment:
We should standardize the returned value whether it's an array with normalized data (contact id, name, email, lists, etc) or just the contact ID.

This PR changes from the standardized `boolean` to ID for ActiveCampaign and an uncontrolled array from the ESP returned data.

Reply:
Done in 9dee2c39232304cf8f410bb0ecbba2209001f55c – an array with `'id'` key will always be returned.